azurerm/internal/services/policy/policy_assignment_resource.go

```diff
@@ -123,10 +126,40 @@ func resourceArmPolicyAssignment() *schema.Resource {
 				Optional: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
+
+			"metadata": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ValidateFunc:     validation.StringIsJSON,
+				DiffSuppressFunc: policyAssignmentsMetadataDiffSuppressFunc,
+			},
```

Comment:
I think this will need to be `Computed` as the builtin policies will set some of this information even if this is omitted in the config.

Reply:
Thanks @jackofallops, i've put `Computed` back